src/main/java/org/breedinginsight/services/parsers/trait/TraitFileParser.java

```diff
@@ -0,0 +1,230 @@
+
+        Workbook workbook = null;
+        try {
+            workbook = WorkbookFactory.create(inputStream);
+        } catch (IOException | EncryptedDocumentException e) {
+            log.error(e.getMessage());
+            throw new ParsingException("Error reading file");
+        }
+
+        Sheet sheet = workbook.getSheet(EXCEL_DATA_SHEET_NAME);
+        if (sheet == null) {
+            throw new ParsingException("Missing sheet" + EXCEL_DATA_SHEET_NAME);
+        }
+
+        ExcelParser parser = new ExcelParser();
+        List<ExcelRecord> records = parser.parse(sheet, TraitFileColumns.getColumns());
+        return excelRecordsToTraits(records);
+    }
+
+    // no sheets RFC4180
+    public List<Trait> parseCsv(@NonNull InputStream inputStream) throws ParsingException {
+
+        ArrayList<Trait> traits = new ArrayList<>();
+        InputStreamReader in = new InputStreamReader(inputStream);
+
+        Iterable<CSVRecord> records = null;
+        try {
+            // withHeader for enum uses name() internally so we have to give string array instead
+            records = CSVFormat.DEFAULT
+                    .parse(in);
+        } catch (IOException e) {
+            log.error(e.getMessage());
+            throw new ParsingException("Error reading file");
+        }
+
+        Sheet excelSheet = convertCsvToExcel(records);
+        ExcelParser parser = new ExcelParser();
+        List<ExcelRecord> excelRecords = parser.parse(excelSheet, TraitFileColumns.getColumns());
+        return excelRecordsToTraits(excelRecords);
+
+    }
+
+    private List<Trait> excelRecordsToTraits(List<ExcelRecord> records) {
+        List<Trait> traits = new ArrayList<>();
+
+        for (ExcelRecord record : records) {
+
+            ProgramObservationLevel level = ProgramObservationLevel.builder()
+                    .name(parseExcelValueAsString(record, TraitFileColumns.TRAIT_LEVEL))
+                    .build();
+
+            // TODO: throw if not active/archived/empty
+            Boolean active = !parseExcelValueAsString(record, TraitFileColumns.TRAIT_STATUS).equals(TRAIT_STATUS_ARCHIVED);
+
+            Method method = Method.builder()
+                    .methodName(parseExcelValueAsString(record, TraitFileColumns.METHOD_NAME))
+                    .description(parseExcelValueAsString(record, TraitFileColumns.METHOD_DESCRIPTION))
+                    .methodClass(parseExcelValueAsString(record, TraitFileColumns.METHOD_CLASS))
+                    .formula(parseExcelValueAsString(record, TraitFileColumns.METHOD_FORMULA))
+                    .build();
+
+            // TODO: throw if not valid datatype
+            DataType dataType = null;
+
+            try {
+                dataType = DataType.valueOf(parseExcelValueAsString(record, TraitFileColumns.SCALE_CLASS).toUpperCase());
+            } catch (IllegalArgumentException e) {
+                log.error(e.getMessage());
+            }
+
+            // TODO: throw if bad format
```

Comment:
There wasn't really anything to do here. The way categories work is <label of the category> = <meaning of the category> but you don't have to specify a label. So there's not really a valid check to do since the = is not required. I added a test though to make sure categories with an empty value are handled properly (this would cover all list types as well. Pushed changes.